packages/snowpack/src/commands/dev.ts

```diff
@@ -127,22 +129,41 @@ const sendFile = (
     acceptEncoding = '';
   }
 
-  function onError(err) {
-    if (err) {
-      res.end();
-      console.error(
-        colors.red(`  ✘ An error occurred while compressing ${colors.bold(req.url)}`),
-        err,
-      );
-    }
-  }
-
+  // Handle gzip compression
   if (/\bgzip\b/.test(acceptEncoding) && stream.Readable.from) {
     const bodyStream = stream.Readable.from([body]);
     headers['Content-Encoding'] = 'gzip';
     res.writeHead(200, headers);
-    stream.pipeline(bodyStream, zlib.createGzip(), res, onError);
-    return;
+    stream.pipeline(bodyStream, zlib.createGzip(), res, function onError(err) {
+      if (err) {
+        res.end();
+        console.error(
+          colors.red(`  ✘ An error occurred while compressing ${colors.bold(req.url)}`),
```

Comment:
+1 for moving this here, lets also make it more generic: "an error occurred serving X"